### ISSUE TITLE
Possible to set ConfirmDialog Synchronous?

### ISSUE THREAD

**Issue by JJarrie** [samie/Vaadin-ConfirmDialog#23]
Hi,

I tried to make a user's validation on each object of a collection (in a loop), but actually ConfirmDialog.show() isn't blocking and the loop finish before to display it.

Have you an issue for this case? Or it's impossible actually?

Thank

(Sorry for my English)

**Comment by samie** [comment]
Yes, you right. The underlying communication is still asynchronous (XHR most likely). So if you want to pipeline them (one after another) you need to chain the callbacks. Or maybe a place to use Java 8 stream APIs?

**Comment by mvysny** [comment]
This is definitely not possible. The Vaadin UI thread cannot block in the show() method - it must finish and deliver the dialog to be shown to the browser.
The only way this could be done is via continuations: https://kotlinlang.org/docs/reference/coroutines.html

**Comment by mvysny** [comment]
Please read this blogpost regarding blocking dialogs: http://mavi.logdown.com/posts/3488105
